File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

input = sys.stdin.readline
N,M = map(int, input().split())
lst = [0]+list(map(int, input().split()))
prefix = [0]*(N+1)
for i in range(1,N+1):
    prefix[i] = prefix[i-1]+lst[i]

for _ in range(M):
    i,j = map(int, input().split())
    print(prefix[j]-prefix[i-1])

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.js
let input = require("fs").readFileSync("dev/stdin").toString().trim().split("\n");
const [N, M] = input[0].split(" ").map(Number);
const lst = input[1].split(" ").map(Number);
const arr = input.slice(2, M + 2).map((el) => el.split(" ").map(Number));

lst.unshift(0);

const prefix = Array(N + 1).fill(0);

for (let i = 1; i < lst.length; i++) {
  prefix[i] = prefix[i - 1] + lst[i];
}

for (let i = 0; i < arr.length; i++) {
  console.log(prefix[arr[i][1]] - prefix[arr[i][0] - 1]);
}
